compose/integrations/composable-test-cases/settings.gradle.kts

```diff
@@ -0,0 +1,130 @@
+    Default("default"),
+    FailingJs("failingJs"),
+    ;
+
+    fun isDefault() = this == Default
+}
+
+val casesToRun = CasesToRun.values().firstOrNull {
+    it.value == extra["tests.casesToRun"]
+} ?: CasesToRun.Default
+
+
+val listOfFailingJsCases = (extra["tests.failing.kjs"] as String).split(",")
+val failingJsSuffix = "failingJs"
+
+fun getFailingSuffix(testCaseName: String): String? {
+    if (casesToRun == CasesToRun.FailingJs &&
+        listOfFailingJsCases.contains(testCaseName)
+    ) {
+        return failingJsSuffix
+    }
+    return null
+}
+
+fun addRememberAnonymousObjTestCase(testFailingJs: Boolean = false) {
+    val libName = ":testcase-rememberAnonymousObj-lib".let {
+        if (testFailingJs) {
+            it.replace("-lib", "-$failingJsSuffix-lib")
+        } else {
+            it
+        }
+    }
+    val mainName = ":testcase-rememberAnonymousObj-main".let {
+        if (testFailingJs) {
+            it.replace("-main", "-$failingJsSuffix-main")
+        } else {
+            it
+        }
+    }
+    module(libName, "testcases/rememberAnonymousObj/lib")
+    module(mainName, "testcases/rememberAnonymousObj/main")
+}
+
+/**
+ * @param name - the name of a test case
+ * @param failingTestCaseNameSuffix - the suffix to add for a failing test case.
+ * Each platform should have its own unique suffix.
+ */
+fun addATestCase(name: String, failingTestCaseNameSuffix: String? = null) {
+    val libName = ":testcase-$name-lib".let {
+        if (failingTestCaseNameSuffix != null) {
+            it.replace("-lib", "-$failingTestCaseNameSuffix-lib")
+        } else {
+            it
+        }
+    }
+    val mainName = ":testcase-$name-main".let {
+        if (failingTestCaseNameSuffix != null) {
+            it.replace("-main", "-$failingTestCaseNameSuffix-main")
+        } else {
+            it
+        }
+    }
+
+    println("adding $libName, $mainName")
+    module(libName, "testcases/$name/lib")
+    module(mainName, "testcases/$name/main")
+}
+
+include(":common")
+
+if (casesToRun.isDefault()) {
+    module(":testcase-template-lib", "testcases/template/lib")
+    module(":testcase-template-main", "testcases/template/main")
+
+    module(":testcase-inheritance-composableInterface-lib", "testcases/inheritance/composableInterface/lib")
+    module(":testcase-inheritance-composableInterface-main", "testcases/inheritance/composableInterface/main")
+
+    module(":testcase-inheritance-funInterface-lib", "testcases/inheritance/funInterface/lib")
+    module(":testcase-inheritance-funInterface-main", "testcases/inheritance/funInterface/main")
+
+    module(":testcase-constructors-lib", "testcases/constructors/lib")
+    module(":testcase-constructors-main", "testcases/constructors/main")
+
+    module(":testcase-anonymousObjects-lib", "testcases/anonymousObjects/lib")
+    module(":testcase-anonymousObjects-main", "testcases/anonymousObjects/main")
+
+    module(":testcase-valueClass-lib", "testcases/valueClass/lib")
```

Comment:
I have 2 failing tests on Windows:
```
> Task :testcase-valueClass-main:compileTestKotlinLinuxX64 FAILED
e: Compilation failed: Sequence is empty.

 * Source files: Tests.kt
 * Compiler version info: Konan: 1.8.10 / Kotlin: 1.8.10
 * Output kind: LIBRARY

e: java.util.NoSuchElementException: Sequence is empty.
        at kotlin.sequences.SequencesKt___SequencesKt.first(_Sequences.kt:112)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.defaultValue(ComposerParamTransformer.kt:261)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.defaultValue$default(ComposerParamTransformer.kt:240)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.defaultArgumentFor(ComposerParamTransformer.kt:224)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.withComposerParamIfNeeded(ComposerParamTransformer.kt:176)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer$copyWithComposerParam$2$3.visitCall(ComposerParamTransformer.kt:555)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitCall(IrElementTransformerVoid.kt:215)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitCall(IrElementTransformerVoid.kt:24)
...
        at org.jetbrains.kotlin.ir.declarations.IrFunction.transformChildren(IrFunction.kt:61)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoidKt.transformChildrenVoid(IrElementTransformerVoid.kt:346)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.copyWithComposerParam(ComposerParamTransformer.kt:508)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.withComposerParamIfNeeded(ComposerParamTransformer.kt:306)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.visitSimpleFunction(ComposerParamTransformer.kt:139)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitSimpleFunction(IrElementTransformerVoid.kt:73)


> Task :testcase-valueClass-main:compileTestKotlinDesktop
w:  `suppressKotlinVersionCompatibilityCheck` is set to the same version of Kotlin that the Compose Compiler was already expecting (Kotlin 1.8.10), and thus has no effect and should be removed.

> Task :testcase-valueClass-main:compileTestKotlinJs FAILED
e: java.util.NoSuchElementException: Sequence is empty.
        at kotlin.sequences.SequencesKt___SequencesKt.first(_Sequences.kt:112)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.defaultValue(ComposerParamTransformer.kt:261)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.defaultValue$default(ComposerParamTransformer.kt:240)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.defaultArgumentFor(ComposerParamTransformer.kt:224)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.withComposerParamIfNeeded(ComposerParamTransformer.kt:176)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer$copyWithComposerParam$2$3.visitCall(ComposerParamTransformer.kt:555)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitCall(IrElementTransformerVoid.kt:215)
...
        at org.jetbrains.kotlin.ir.declarations.IrFunction.transformChildren(IrFunction.kt:61)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoidKt.transformChildrenVoid(IrElementTransformerVoid.kt:346)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.copyWithComposerParam(ComposerParamTransformer.kt:508)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.withComposerParamIfNeeded(ComposerParamTransformer.kt:306)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.visitSimpleFunction(ComposerParamTransformer.kt:139)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitSimpleFunction(IrElementTransformerVoid.kt:73)
...
        at org.jetbrains.kotlin.ir.util.TransformKt.transformInPlace(transform.kt:35)
        at org.jetbrains.kotlin.ir.declarations.IrModuleFragment.transformChildren(IrModuleFragment.kt:51)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoidKt.transformChildrenVoid(IrElementTransformerVoid.kt:346)
        at androidx.compose.compiler.plugins.kotlin.lower.ComposerParamTransformer.lower(ComposerParamTransformer.kt:105)
        at androidx.compose.compiler.plugins.kotlin.ComposeIrGenerationExtension.generate(ComposeIrGenerationExtension.kt:153)
        at org.jetbrains.kotlin.ir.backend.js.KlibKt.generateModuleFragmentWithPlugins$lambda$20(klib.kt:552)
        at org.jetbrains.kotlin.psi2ir.Psi2IrTranslator.generateModuleFragment(Psi2IrTranslator.kt:104)
```

Reply:
I forgot to update Readme.

Could you please try the latest version
compose.kotlinCompilerPluginVersion=1.4.2-rc03

Reply:
It works now!